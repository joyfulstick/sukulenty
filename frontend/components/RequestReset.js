import React, { Component } from 'react'
import Error from './ErrorMessage'
import Form from './styles/Form'
import { Mutation } from 'react-apollo'
import gql from 'graphql-tag'

const REQUEST_RESET_MUTATION = gql`
  mutation REQUEST_RESET_MUTATION($email: String!) {
    requestReset(email: $email) {
      message
    }
  }
`

class RequestReset extends Component {
  state = {
    email: '',
  }

  handleChangeValue = e => this.setState({ [e.target.name]: e.target.value })

  render() {
    const { state, handleChangeValue } = this
    return (
      <Mutation mutation={REQUEST_RESET_MUTATION} variables={state}>
        {(requestReset, { error, loading, called }) => (
          <Form
            method="post"
            onSubmit={async e => {
              e.preventDefault()
              await requestReset().catch(err => console.log(err)) // eslint-disable-line
              this.setState({
                email: '',
              })
            }}
            data-test="form"
          >
            <fieldset disabled={loading} aria-busy={loading}>
              <h2>Resetowanie has≈Ça</h2>
              <Error error={error} />
              {!error &&
                !loading &&
                called && (
                  <p>
                    Wiadomo≈õƒá z linkiem do resetu has≈Ça zosta≈Ça wys≈Çana. Sprawd≈∫
                    skrzynkƒô e-mail üì¨
                  </p>
                )}
              <label>
                Email
                <input
                  type="email"
                  name="email"
                  placeholder="e-mail"
                  value={state.email}
                  onChange={handleChangeValue}
                  required
                />
              </label>
              <button type="submit">Resetuj</button>
            </fieldset>
          </Form>
        )}
      </Mutation>
    )
  }
}

export { RequestReset as default, REQUEST_RESET_MUTATION }
