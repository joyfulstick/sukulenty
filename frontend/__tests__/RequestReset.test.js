import RequestReset, {
  REQUEST_RESET_MUTATION,
} from '../components/RequestReset'
import { MockedProvider } from 'react-apollo/test-utils'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import wait from 'waait'

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: 'test@test.pl' },
    },
    result: {
      data: { requestReset: { message: 'success', __typename: 'Message' } },
    },
  },
]

describe('<RequestReset/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>,
    )
    const form = wrapper.find('form[data-test="form"]')
    expect(toJSON(form)).toMatchSnapshot()
  })

  it('calls the mutation', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>,
    )
    wrapper.find('input').simulate('change', {
      target: { name: 'email', value: 'test@test.pl' },
    })
    wrapper.find('form').simulate('submit')
    await wait(50)
    wrapper.update()
    expect(wrapper.find('p').text()).toContain(
      'WiadomoÅ›Ä‡ z linkiem do resetu hasÅ‚a zostaÅ‚a wysÅ‚ana. SprawdÅº skrzynkÄ™ e-mail ðŸ“¬',
    )
  })
})
