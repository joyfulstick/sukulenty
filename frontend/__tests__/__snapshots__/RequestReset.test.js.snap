// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> renders and matches snapshot 1`] = `
<form
  className="Form-t0rxhl-0 lpjQDY"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Resetowanie has≈Ça
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label>
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="e-mail"
        required={true}
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Resetuj
    </button>
  </fieldset>
</form>
`;
